string/allocated_string.h

```diff
@@ -0,0 +1,23 @@
+//
+// Created by aaron on 8/19/22.
+//
+
+#ifndef LAUNCH_CORE_ALLOCATED_STRING_H
+#define LAUNCH_CORE_ALLOCATED_STRING_H
+
+
+#include <cstddef>
+#include "string.h"
+
+namespace string {
+    template <size_t size>
+class String : public ::String<size> {
+public:
+    String(char const *buffer, const size_t str_size) {
```

Comment:
you should probably flip this constructor with the one in ::String, the use case is going to be that you want to pass a "String" type around without templated parameters, but when we declare a string we need to specify a size so we make it some type that inherits from string. Look at _pool/pool.h_ for an example.

You can alternatively make a string interface (sort of like ETL's istring) and inherit from that, so you would pass around istrings as parameters and the actual declarations are templated strings.